src/components/draw/DrawDirective.js

```diff
@@ -132,28 +285,72 @@ goog.require('ga_map_service');
             if (layer.adminId) {
               updateShortenUrl(layer.adminId);
             }
-
-            if (lastActiveTool) {
-              activateTool(lastActiveTool);
+            ol.Observable.unByKey(unLayerAdd);
+            ol.Observable.unByKey(unLayerRemove);
+            while (unWatch.length) {
+              unWatch.pop()();
             }
-
-            unLayerRemove = map.getLayers().on('remove', function(evt) {
-              if (evt.element === layer) {
+            // if a layer is added from other component (Import KML, Permalink,
+            // DnD ...) and the currentlayer has no features, we define a
+            // new layer.
+            unLayerAdd = map.getLayers().on('add', function(evt) {
+              if (gaMapUtils.isStoredKmlLayer(evt.element) &&
+                  layer.getSource().getFeatures().length == 0 &&
+                  !useTemporaryLayer) {
                 defineLayerToModify();
               }
             });
+
+            // Block dblclick event in draw mode to avoid map zooming
+            unDblClick = map.on('dblclick', function(evt) {
+              return false;
+            });
+            select.setActive(true);
+
+            // Active watchers
+            // Update selected feature's style when the user change a value
+            unWatch.push(scope.$watchGroup([
+              'options.icon',
+              'options.iconSize',
+              'options.color',
+              'options.name',
+              'options.description'
+            ], function() {
+              if (select.getActive()) {
+                var feature = select.getFeatures().item(0);
+                if (feature) {
+                  // Update the style of the feature with the current style
+                  var styles = scope.options.updateStyle(feature);
+                  feature.setStyle(styles);
+                  // then apply the select style
+                  styles = scope.options.selectStyleFunction(feature);
+                  feature.setStyle(styles);
+                }
+              }
+            }));
+
+            unWatch.push(scope.$watch('popupToggle', function(active) {
```

Comment:
I remnber now. I need to watch when the user click on close button of the popup.